libmamba/src/solv-cpp/repo.cpp

```diff
@@ -83,34 +83,74 @@ namespace mamba::solv
 
     namespace
     {
-        struct FilePtr
+        class FilePtr
         {
-            std::FILE* ptr = nullptr;
-            std::string_view name = {};
+        public:
 
-            ~FilePtr() noexcept(false)
+            /**
+             * Open a file with C API.
+             *
+             * @param path must hae filesystem default encoding.
+             */
+            static auto open(std::filesystem::path path, const char* mode) -> FilePtr;
```

Comment:
Why not use a constructor and make this object testable instead (with `is_open() const` and `explicit operator bool() const`? Basically like in the standard. (thats also whatś in powerloader if my memory is correct)